src/NumericalAlgorithms/Integration/GslQuadAdaptive.hpp

```diff
@@ -24,7 +25,7 @@ namespace detail {
 // passed as a void* and re-interpreted appropriately.
 template <typename IntegrandType>
 double integrand(const double x, void* const params) {
-  const auto function = reinterpret_cast<IntegrandType*>(params);
+  const auto function = reinterpret_cast<IntegrandType* const>(params);
```

Comment:
While investigating this I found that you are not using Gsl.hpp, and you are missing an include for std::vector in the header.  (Didn't check the cpp or test.)